### PR TITLE
repo-sync-2024-01-16T11:24:45+0800

### PR DESCRIPTION
**Pull request by zhouaihui**
Weekly sync

### Added
- sgb/ss_xgb/ss_glm/ss_sgd can save specified columns with result now.
- ss_glm: expose two params infeed size and newton.
- SplitLearning
  - Add a defense method against data reconstruction attacks.
  - Adapt replay attack and gradreplace attack for deepfm.
  - Add expoit attack.

### Changed
- Component: update fillna descriptions.

### Fixed
- Fix SS-GLM toturial.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -29,7 +29,7 @@ setuptools>=65.5.1  # FEATURE=[lite]
 # For security reason.
 certifi>=2023.7.22  # FEATURE=[lite]
 sparse>=0.14.0
-spu==0.7.0.dev20231222  # FEATURE=[lite]
+spu==0.7.0.dev20231229  # FEATURE=[lite]
```
Comment (anakinxc) [inline requirements.txt:32]: 0.8.0.dev20240112

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  - Adapt replay attack and gradreplace attack for deepfm.
+  - Add expoit attack.
+
+### Changed
+- Component: update fillna descriptions.
+
+### Fixed
+- Fix SS-GLM toturial.
+
 ## [1.4.0.dev240103] - 2024-1-3
```
Comment (fy222fy) [inline CHANGELOG.md:31]: 这个文件好像没有正确地同步回来。 只有这个的话问题也不大

File: docs/getting_started/debug_mode.md [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+隐语框架有一个很重要的特点就是:**用户只需要写一套secretflow的代码，可以在多种部署条件下直接执行！**  
+这意味着，用户可以使用简单易部署的方式进行开发，随后**零成本**的迁移到生产环境产生价值。  
 我们的隐语目前已经提供了两种模式分别是：
 - Simulation mode
 - Producation mode
 
-Simulation mode 是使用单一的ray集群来进行调度，可以部署在单机上，也可以部署在多机环境，部署和使用难度都相对比较低。
-Production mode 则是使用多个ray集群来进行调度，借助隐语和Ray社区合作开发的RayFed进行安全的数据交换。来解决生产环境数据安全问题。
+Simulation mode 是使用单一的ray集群来进行调度，可以部署在单机上，也可以部署在多机环境，部署和使用难度都相对比较低。  
+Production mode 则是使用多个ray集群来进行调度，借助隐语和Ray社区合作开发的RayFed进行安全的数据交换。来解决生产环境数据安全问题。
```
Comment (fy222fy) [inline docs/getting_started/debug_mode.md:9]: 更改了好多这个行位空格，看看是咋回事？ 内部black带出来的？

File: CHANGELOG.md (hunk 2 above)
Comment (zhouaihui) [reply CHANGELOG.md:31]: 同步回来是后续操作。

File: docs/getting_started/debug_mode.md (hunk 3 above)
Comment (zhouaihui) [reply docs/getting_started/debug_mode.md:9]: Nice catch!

**Review by fy222fy — APPROVED** [review]

**Pull request merged**